packages/vue/src/components/FCrudDataset/FCrudDataset.vue

```diff
@@ -1,3 +1,307 @@
+        default(): void {
+            /* do nothing */
+        },
+    },
+    /**
+     * If given, this function is called before the form data is validated and the [[submit]] event is emitted.
+     * See <f-validation-form> `beforeValidation` props for more info.
+     */
+    beforeValidation: {
+        type: Function as PropType<FValidationFormCallback>,
+        required: false,
+        default(): void {
+            /* do nothing */
+        },
+    },
+    /**
+     * If given, this function is called after the modal has been closed.
+     */
+    onCancel: {
+        type: Function as PropType<() => void>,
+        required: false,
+        default() {
+            return undefined;
+        },
+    },
+
+    /**
+     * Property for changing the "add new" modal heading
+     */
+    addNewModalHeader: {
+        type: String,
+        required: false,
+        default: TranslationService.provider.translate("fkui.crud-dataset.modal.header.add", "Lägg till rad"),
+    },
+    /**
+     * Property for changing the "modify" modal heading
+     */
+    modifyModalHeader: {
+        type: String,
+        required: false,
+        default: TranslationService.provider.translate("fkui.crud-dataset.modal.header.modify", "Ändra rad"),
+    },
+    /**
+     * Property for changing the "delete" modal heading
+     */
+    deleteModalHeader: {
+        type: String,
+        required: false,
+        default: TranslationService.provider.translate(
+            "fkui.crud-dataset.modal.header.delete",
+            "Är du säker på att du vill ta bort raden?",
+        ),
+    },
+});
+
+const emit = defineEmits(["created", "deleted", "updated", "update:modelValue"]);
+
+const formModalButtons = computed((): FModalButtonDescriptor[] => {
+    const confirmButtonText =
+        operation.value === Operation.ADD
+            ? $t("fkui.crud-dataset.modal.confirm.add", "Lägg till")
+            : $t("fkui.crud-dataset.modal.confirm.modify", "Spara");
+    const cancelButtonText =
+        operation.value === Operation.ADD
+            ? $t("fkui.crud-dataset.modal.cancel.add", "Avbryt")
+            : $t("fkui.crud-dataset.modal.cancel.modify", "Avbryt");
+    return [
+        {
+            label: confirmButtonText,
+            event: "confirm",
+            type: "primary",
+            submitButton: true,
+        },
+        {
+            label: cancelButtonText,
+            event: "dismiss",
+            type: "secondary",
+            submitButton: false,
+        },
+    ];
+});
+
+const confirmDeleteButtons = computed((): FModalButtonDescriptor[] => {
+    return [
+        {
+            label: $t("fkui.crud-dataset.modal.confirm.delete", "Ja, ta bort"),
+            type: "primary",
+            event: "confirm",
+        },
+        {
+            label: $t("fkui.crud-dataset.modal.cancel.delete", "Nej, avbryt"),
+            type: "secondary",
+        },
+    ];
+});
+
+const hasAddSlot = computed((): boolean => {
```

Comment:
(behövs det anges typer för computed-metoderna?)

Reply:
Oklart, det fungerar att bygga och jag tycker det borde fungera utan men i editorn får man fel:

```
'slots' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
```

```
'hasAddSlot' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
```

```
Function implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.
```

Reply:
Låter det vara så tills vidare, hittar vi orsaken till varför vscode ger felen ovan så kan vi rätta det senare